addOns/graphql/src/test/java/org/zaproxy/addon/graphql/GraphQlParserUnitTest.java

```diff
@@ -70,4 +79,23 @@ void shouldFailIntrospectionWhenResponseDataIsNull() throws Exception {
         // When/Then
         assertThrows(IOException.class, gqp::introspect);
     }
+
+    @Test
+    void shouldRaiseAlertWhenSpecified() throws Exception {
+        // Given
+        String introspectionResponse =
+                "{\"data\":{\"__schema\":{\"queryType\":{\"name\":\"Root\"},\"types\":[{\"kind\":\"OBJECT\",\"name\":\"Root\",\"fields\":[{\"name\":\"zap\",\"args\":[],\"type\":{\"kind\":\"SCALAR\",\"name\":\"String\"}}]}]}}}";
+        nano.addHandler(new StaticContentServerHandler("/", introspectionResponse));
+        GraphQlParser gqp = new GraphQlParser(endpointUrl);
+        var extAlert = mock(ExtensionAlert.class);
+        Control.getSingleton().getExtensionLoader().addExtension(extAlert);
+        // When
+        gqp.introspect(true);
+        // Then
+        var alert = ArgumentCaptor.forClass(Alert.class);
+        verify(extAlert).alertFound(alert.capture(), any());
+        assertThat(alert.getValue().getPluginId(), is(ExtensionGraphQl.TOOL_ALERT_ID));
+        assertThat(alert.getValue().getAlertRef(), is(ExtensionGraphQl.TOOL_ALERT_ID + "-1"));
+        assertThat(alert.getValue().getName(), is("!graphql.introspection.alert.name!"));
```

Comment:
Wouldn’t this mean the resource didn’t load properly?

Reply:
Yes, but it's only for the test.

Reply:
Okay